BorrowMyProducts/src/test/java/com/techhunters/borrowmyproducts/daorepository/ProductRepositoryTest.java

```diff
@@ -0,0 +1,121 @@
+package com.techhunters.borrowmyproducts.daorepository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.techhunters.borrowmyproducts.entity.Category;
+import com.techhunters.borrowmyproducts.entity.Product;
+import com.techhunters.borrowmyproducts.entity.User;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class ProductRepositoryTest {
+    
+	 
+	    @Autowired
+	    private ProductRepository productRepository;
+	    
+	    @Autowired 
+	    private CategoryRepository categoryRepository;
```

Comment:
why are other repositories required for ProductRepository test?

Reply:
Hi,
In order to assign a user to a product,the user should already be present in database.So,I used other repositories to save related items.I think this can be resolved by applying  cascading(persist type) from product to related entities.I haven't applied cascading yet,so I had to use these other repositories for now.

Reply:
updated it